src/sdk/openmldb_api.h

```diff
@@ -0,0 +1,215 @@
+
+// TimeStamp corresponds to TIMESTAMP in SQL, as a parameter inserted into ParameterRow or RequestRow
+// constructor : only one parameter of type int_64
+class TimeStamp {
+ public:
+    explicit TimeStamp(int64_t _val) : value(_val) {}
+    ~TimeStamp() {}
+    int64_t get_Timestamp() const;
+
+ private:
+    int64_t value;
+};
+
+// Date corresponds to DATE in SQL, as a parameter inserted into ParameterRow or RequestRow
+// constructor : first parameter           int_32    year
+//                    second parameter      int_32    month
+//                    third parameter          int_32    day
+class Date {
+ public:
+    Date(int32_t _year, int32_t _month, int32_t _day) : year(_year), month(_month), day(_day) {}
+    ~Date() {}
+    int32_t get_year() const;
+    int32_t get_month() const;
+    int32_t get_day() const;
+
+ private:
+    int32_t year;
+    int32_t month;
+    int32_t day;
+};
+
+// OpenmldbNull corresponds to NULL in SQL, as a parameter inserted into ParameterRow or RequestRow
+// constructor : no parameters
+class OpenmldbNull {
+ public:
+    OpenmldbNull() {}
+};
+
+// OpenmldbHandler is used to link openmldb server. to use openmldb, you must create an object of this type
+// cluster :
+// constructor : first parameter         string      format : "IP:Port"    eg : "127.0.0.1:2181"
+//                    second parameter    string      format : "path"       eg : "/openmldb"
+// standalone :
+// constructor : first parameter         string      format : "IP"           eg : "127.0.0.1"
+//                    second parameter    string      format : "Port"        eg : 6527
+class OpenmldbHandler {
+ public:
+    OpenmldbHandler(std::string _zk_cluster, std::string _zk_path);
+    OpenmldbHandler(std::string _host, uint32_t _port);
+    ~OpenmldbHandler();
+    openmldb::sdk::SQLClusterRouter* get_router() const;
+    hybridse::sdk::Status* get_status() const;
+
+ private:
+    OpenmldbHandler(const OpenmldbHandler&);
+    OpenmldbHandler& operator=(const OpenmldbHandler&);
+
+ private:
+    openmldb::sdk::SQLRouterOptions* cluster = nullptr;
+    openmldb::sdk::StandaloneOptions* standalone = nullptr;
+    openmldb::sdk::SQLClusterRouter* router = nullptr;
+    hybridse::sdk::Status* status = nullptr;
+};
+
+// In the request with parameter and request mode of openmldb, the position of parameters to be filled is indicated by
+// the symbol "?" to express
+
+// ParameterRow is used to create a parameter line. insert data into objects of type ParameterRow, and replace the "?"
+// constructor : only one parameter of type OpenmldbHandler*
+// insert data into objects of type ParameterRow using operator '<<'     eg : para << 1 << "hello";
+// reset() is used to clear the inserted data in an object of type ParameterRow
+class ParameterRow {
+ public:
+    explicit ParameterRow(const OpenmldbHandler* _handler);
+    std::shared_ptr<openmldb::sdk::SQLRequestRow> get_parameter_row() const;
```

Comment:
do not expose the function that user will not use

Reply:
But execute_parameterized need to use it, so keep it